### ISSUE TITLE
MLflow server crashes when serving parameters with spaces in the name

### ISSUE THREAD

**Issue by giordyb** [mlflow/mlflow#249]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
NO
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Python:3 docker image
- **MLflow installed from (source or binary)**: 
binary (pip)
- **MLflow version (run ``mlflow --version``)**:
0.4.1
- **Python version**: 
3.6
- **npm version (if running the dev UI):
- **Exact command to reproduce**:

### Describe the problem
since 0.4.1 mlflow server crashes with an "internal server error" when the run contains a parameter with spaces (i.e. "train times").

this is the error that gets logged

```
[2018-08-06 09:59:26,971] ERROR in app: Exception on /ajax-api/2.0/preview/mlflow/artifacts/list [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.6/site-packages/mlflow/server/handlers.py", line 178, in _list_artifacts
    run = _get_store().get_run(request_message.run_uuid)
  File "/usr/local/lib/python3.6/site-packages/mlflow/store/file_store.py", line 187, in get_run
    params = self.get_all_params(run_uuid)
  File "/usr/local/lib/python3.6/site-packages/mlflow/store/file_store.py", line 278, in get_all_params
    params.append(self._get_param_from_file(parent_path, param_file))
  File "/usr/local/lib/python3.6/site-packages/mlflow/store/file_store.py", line 257, in _get_param_from_file
    _validate_param_name(param_name)
  File "/usr/local/lib/python3.6/site-packages/mlflow/utils/validation.py", line 25, in _validate_param_name
    raise Exception("Invalid parameter name: '%s'" % name)
Exception: Invalid parameter name: 'train times'
```

if I rename the parameter to "train_times" or I revert back to 0.4.0 then the experiment gets displayed correctly.

**Comment by smurching** [comment]
@giordyb thanks for reporting this issue, it's caused by param/metric-name validation added in 0.4.1. #242, which was included in MLflow 0.4.2, should fix this issue. Would you be able to try upgrading to 0.4.2 & rerunning? Thanks!

**Comment by smurching** [comment]
Closing this issue for now since it's a known bug / was fixed in 0.4.2, but please reopen if the suggested fix doesn't work out - thanks @giordyb!
